File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    TreeNode* lowestCommonAncestor(TreeNode* root, TreeNode* p, TreeNode* q) {

        if(p->val>q->val){
            return lowestCommonAncestor(root,q,p);
        }

        int sm = p->val;
        int hi = q->val;

        if (sm <= root->val && root->val <= hi){
            return root;
        }

        if(root->val>sm){
            return lowestCommonAncestor(root->left,q,p);
        }
        if(root->val<hi){
            return lowestCommonAncestor(root->right,q,p);
        }

        return nullptr;
        
        
    }
};

File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorder(self, root: Optional[TreeNode],ans:List[int]):
        if root==None:
            return 
        self.inorder(root.left,ans)
        ans.append(root.val)
        self.inorder(root.right,ans)
        return 

    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        ans=[]
        self.inorder(root,ans)
        return ans
        